azure_pipeline-global.yaml

```diff
@@ -0,0 +1,335 @@
+    - /environments/ithc/ithc.tfvars
+    - /environments/ldata/ldata.tfvars
+    - /environments/prod/prod.tfvars
+    - /environments/ptl/ptl.tfvars
+    - /environments/sbox/sbox.tfvars
+    - /environments/stg/stg.tfvars
+    - /environments/test/test.tfvars
+    
+variables:
+  terraformVersion: 0.12.26
+  agentPool: 'hmcts-agent-pool'
+  action:
+
+stages:
+  - stage: Precheck
+    jobs:
+      - job: Validate
+        pool: ${{ variables.agentPool }}
+        steps:
+        - task: AzureKeyVault@1
+          displayName: Retrieve keyvault secret for ADO token
+          inputs:
+            ConnectedServiceName: azurerm-sandbox
+            keyVaultName: infra-vault-nonprod
+            secretsFilter: 'azure-devops-token'
+            runAsPreJob: false
+        - task: Bash@3
+          displayName: Prevent parallel run
+          env:
+            thisbuild: $(Build.BuildId)
+            pipelinedefinition: $(System.DefinitionId)
+          inputs:
+            targetType: 'inline'
+            script: | 
+              #!/bin/bash
+              set -e
+              organization="hmcts"
+              project="CNP"
+
+              echo "This is build $thisbuild"
+              IFS=$'\n'
+              JSON_DATA=($(curl -s -u :$(azure-devops-token) --request GET "https://dev.azure.com/$organization/$project/_apis/build/builds?api-version=5.1&definitions=$pipelinedefinition" -H "Content-Type: application/json" | jq  '.value[] | .status + (.id|tostring)' | sort -u | grep inProgress))
+              buildnumber=(${JSON_DATA//[!0-9]/})
+
+
+              if [ $thisbuild -eq $buildnumber ] 
+              then
+
+              echo "No other builds are in Progress"
+
+              else
+
+              while [[ $thisbuild -ne $buildnumber_inprogress ]]
+              do 
+
+              IFS=$'\n'
+              JSON_CHECK=($(curl -s -u :$(azure-devops-token) --request GET "https://dev.azure.com/$organization/$project/_apis/build/builds?api-version=5.1&definitions=$pipelinedefinition" -H "Content-Type: application/json" | jq  '.value[] | .status + (.id|tostring)' | sort -u | grep inProgress))
+              buildnumberrunning=(${JSON_CHECK//[!0-9]/})
+              echo "Build number: $buildnumberrunning is in progress"
+
+              buildnumber_inprogress=$buildnumberrunning
+              sleep 30
+
+              done
+
+              fi
+              
+              echo "Starting $thisbuild"
+
+  - stage: GetArtifacts
+    dependsOn: Precheck
+    jobs:
+      - job: Validate
+        pool: ${{ variables.agentPool }}
+        steps:
+          - task: TerraformInstaller@0
+            inputs:
+              terraformVersion: ${{ variables.terraformVersion }}
+          - task: PowerShell@2
+            displayName: Run tests
+            inputs:
+              targetType: 'inline'
+              script: |
+                Install-Module -Name Pester -MaximumVersion 4.99.99 -Force -Verbose -Scope CurrentUser
```

Comment:
should be a script so we aren't duplicating this ideally

Reply:
I didn't add this to a script as it was only a couple of lines. Can add if you want